app/components/Review/widgets/FileWidget.tsx

```diff
@@ -1,22 +1,144 @@
+import React, { useMemo } from 'react';
 import { WidgetProps } from '@rjsf/core';
+import styled from '@emotion/styled';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
 import DownloadLink from 'components/DownloadLink';
-import React, { useMemo } from 'react';
+import { DateTime } from 'luxon';
+
+const StyledFile = styled('div')`
+  display: flex;
+  flex-direction: column;
+  margin-bottom: 8px;
+`;
+
+const StyledDetails = styled.span`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  color: #939393;
+`;
+
+const StyledSingleDetails = styled(StyledDetails)`
+  justify-content: flex-end;
+`;
+
+const StyledIconDiv = styled('div')`
+  position: relative;
+  & svg {
+    max-width: 14px;
+    position: relative;
+    top: 2px;
+    left: -16px;
+  }
+
+  & a {
+    position: relative;
+    left: -16px;
+  }
+`;
+
+const FileWidget: React.FC<WidgetProps> = ({
+  formContext,
+  id,
+  options,
+  value,
+}) => {
+  // seems like the id is the only way to get the field name from widget props
+  // id is in the format root_[pageName]_[fieldName]
+  const fieldName = id?.split('_')?.[2];
+  const rfiList = formContext?.rfiList;
+  const isRfi = rfiList?.length > 0;
 
-const FileWidget: React.FC<WidgetProps> = ({ value }) => {
   const filesArray = useMemo(() => {
     return value || [];
   }, [value]);
 
-  return filesArray?.map((el, index) => (
-    <React.Fragment key={el.name}>
-      <DownloadLink uuid={el.uuid} fileName={el.name ?? el.toString()} />
-      {index < filesArray.length - 1 && (
-        <>
-          ,<br />
-        </>
+  return (
+    <div>
+      {rfiList?.map((rfi, i) => {
+        // loop through the rfis and check if there are any files for the current field
+        const rfiFiles = rfi.jsonData.rfiAdditionalFiles;
+        const isFileField =
+          rfiFiles && Object.keys(rfiFiles).includes(fieldName);
+        if (!isFileField) return null;
+
+        const attachments = rfi?.attachments?.nodes;
+
+        return (
+          <StyledFile key={rfi}>
+            {rfiFiles[fieldName]?.map((el, index) => {
+              // loop through the list of files for the current field
+              const isSingleFile = !options?.allowMultipleFiles;
+              const isDisplayIcon = isSingleFile && i === 0;
```

Comment:
What's the purpose of the i === 0? Wouldn't this prevent the latest RFI to have the star? Something like this? 
<img width="1040" alt="image" src="https://github.com/bcgov/CONN-CCBC-portal/assets/61289255/160ace45-68a7-4ef9-a217-698ffdbdeba4">

Reply:
In the graphql query we are sorting them by `orderBy: RFI_DATA_ID_DESC` and the `i === 0` is so we display the icon for the first rfi (and newest rfi) that we map. 

The files below the rfis are mapped separately so I am confused as to what you mean. I just double checked and it's working for me:

<img width="1114" alt="Screenshot 2023-08-11 at 9 27 56 AM" src="https://github.com/bcgov/CONN-CCBC-portal/assets/14259474/dfe4d573-2eb4-4f12-b49b-9c2f8aa1a5b1">

Reply:
Sorry! This makes sense now, I was looking at it from the wrong way.

Reply:
No worries thanks 🙂